tape/src/main/java/com/squareup/tape/AnomalyHandlingConfig.java

```diff
@@ -0,0 +1,71 @@
+package com.squareup.tape;
+
+import static com.squareup.tape.TapeLogger.NO_LOG;
+
+/**
+ * This class represents the configuration chosen for how the FileObjectQueue will handle anomalies
+ * that it discovers. It also carries an instance of a [TapeLogger] (if desired) which will log the
+ * events discovered.
+ */
+public class AnomalyHandlingConfig {
+
+  public final AnomalyResponse onReadLargeElementDuringInit;
+  public final AnomalyResponse onReadLargeElement;
+  public final AnomalyResponse onAddLargeElement;
+  public final TapeLogger logger;
+  public final String label;
+
+  /**
+   * Construct a [HandlingConfig]
+   *
+   * @param onReadLargeElementDuringInit - Response when an element > maxElementSize or >
+   *  * fileContentLength is read from an existing file into a FileObjectQueue (e.g. after restart).
+   * @param onReadLargeElement - Response when an element > maxElementSize or >
+   *  * fileContentLength is read during a FileObjectQueue session.
+   * @param onAddLargeElement - Response when an element > maxElementSize is added to the
+   *  * FileObjectQueue.
+   * @oaram label - String to be added to each log or exception.
+   * @param logger - The [TapeLogger] used to log errors above.
+   */
+  public AnomalyHandlingConfig(
+      AnomalyResponse onReadLargeElementDuringInit,
+      AnomalyResponse onReadLargeElement,
+      AnomalyResponse onAddLargeElement,
+      String label,
+      TapeLogger logger
+  ) {
+    this.onReadLargeElementDuringInit = onReadLargeElementDuringInit;
+    this.onReadLargeElement = onReadLargeElement;
+    this.onAddLargeElement = onAddLargeElement;
+    this.label = label;
+    this.logger = logger;
+  }
+
+  /**
+   * Constructor for defaults.
+   *
+   * Default onReadLargeElementDuringInit is AnomalyResponse.LOG.
+   * Default onReadLargeElement is AnomalyResponse.LOG.
+   * Default onAddLargeElement is AnomalyResponse.LOG.
+   * Default label is "TapeQueue"
+   * Default logger is TapeLogger.NO_LOG, which is a no-op.
+   */
+  public AnomalyHandlingConfig() {
+    onReadLargeElementDuringInit = AnomalyResponse.LOG;
+    onReadLargeElement = AnomalyResponse.LOG;
+    onAddLargeElement = AnomalyResponse.LOG;
+    logger = NO_LOG;
+    label = "TapeQueue";
+  }
+
+  /**
+   * THROW throws an Exception.
+   * LOG logs to TapeLogger included with AnomalyHandlingConfig.
+   * IGNORE Ignores problems when adding or reading.
+   */
+  enum AnomalyResponse {
```

Comment:
All of these can be supported by the logger lambda interface if its single abstract method is changed to accept the various data instead of an opaque message. Then the lambda can decide whether to throw, log, ignore, or do something else. It eliminates the need for this enum and should eliminate the need for the label since the lambda supplied by the user can own it. It also eliminates all the conditional API inside the library since you can now unconditionally call into the lambda and let it figure out what to do.

Reply:
Snazzy. Though it deserves a rename in that case as it's more than a logger at that point.

Reply:
Indeed! I keep meaning to write a blog post about this.

My hypotheses is that every time you reach for a logger callback in a library you really want callbacks for observability into the underlying system where logging is one possible action you might want to take.

Reply:
Neat. That's a cool alternative to "every lib ships a logger"